src/adams_methods.jl

```diff
@@ -0,0 +1,626 @@
+                err_kp1, next_dt, timeout_VS = stepsize_hw92!(dt,tdir,y_n,y_np1,
+                                            yerr_kp1, num_stages, timeout_VS,dof,
+                                            abstol, reltol, maxstep, norm)
+
+                # Hairer equation 7.4: If ||LE_{k+1}|| =< 1
+                # then step size is accepted and use next_dt for next step
+                # Else, then step size is rejected, reattempt with fac*next_dt
+                if err_kp1 <= 1
+                    successful_step = true
+                else
+                    successful_step = false
+                    dt = fac*next_dt
+                end
+            end # End of error check for current step size dt
+        end # End of loop for finding successfull step size for nth step
+            # repeats if success_step == false
+
+        #Keep step size constant if `dt` is close to `prev_dt`
+        if abs(next_dt/prev_dt)<2 && abs(next_dt/prev_dt)>=1
+            consec_const_steps += 1
+            next_dt = prev_dt
+        else
+            consec_const_steps = 0
+        end
+
+        #Don't step past `tfinal`
+        if tdir*(t_np1 + next_dt) >= tdir*(tfinal)
+            next_dt = tfinal-t_np1
+        end
+        push!(t,copy(t_np1))
+
+        #--#####################################################################
+        # Interpolation for dense output
+        #--#####################################################################
+
+        # interpolate specified points
+        # TODO: less expensive interpolation. Core problem is that current
+        # frame work does not allow for inexpensive interpolation for uneven
+        # step sizes.
+
+        while iter <= length(tspan) && tdir*tspan[iter]<tdir*t_np1
+            dt_iter = tspan[iter]-t[n]
+
+            # Calculation of g_{j}(n)coefficients for j=0,...,k+1;
+            g_coefs!(g,c,dt,t,tspan[iter],n,min(n,num_stages+1))
+
+            # Calculation of  ϕstar_{j}(n), and ϕ_{j}(n) coef. for j=0,...,k;
+            ϕ_and_ϕstar_coefs!(ϕstar_n, ϕ_n, ϕstar_nm1, b,dy_n,t,tspan[iter],n,1:min(max(n-1,1),num_stages+1))
+
+            yout_iter = copy(y_n)
+            for j = 0:num_stages-1
+                yout_iter += dt_iter*g[(j)+1]*ϕstar_n[(j)+1]
+            end
+
+            # CE iteration for more precise interpolation
+            if n>=num_stages+1
+                dy_iter=F(tspan[iter],yout_iter)
+                ϕ_coefs!(ϕ_np1, ϕstar_n,dy_iter,1:(num_stages)+1)
+
+                yout_iter += dt_iter*g[(num_stages)+1]*ϕ_np1[(num_stages)+1]
+            end
+
+            push!(tout,tspan[iter])
+            push!(yout, copy(yout_iter))
+            iter += 1
+        end
+
+        if iter <= length(tspan) && tspan[iter] == t_np1
+            # handling when tspan and choosen time steps overlap
+            push!(tout, copy(t_np1))
+            push!(yout, copy(y_np1))
```

Comment:
again with the `copy` vs `deepcopy`. Either restrict the function for `Vector` (or `Array`, have you tried it on `Array`? It might already work), or have a path for `deepcopy`.